weblab_cg/_generator_chaste_cg.py

```diff
@@ -0,0 +1,627 @@
+            except AssertionError:
+                pass  # Unit already exists, but that is not a problem
+
+    def _get_membrane_voltage_var(self):
+        """ Find the membrane_voltage variable"""
+        return self._model.get_symbol_by_ontology_term(self._OXMETA, "membrane_voltage")
+
+    def _get_cytosolic_calcium_concentration_var(self):
+        """ Find the cytosolic_calcium_concentration variable if it exists"""
+        try:
+            return self._model.get_symbol_by_ontology_term(self._OXMETA, "cytosolic_calcium_concentration")
+        except KeyError:
+            self._logger.debug(self._model.name + ' has no cytosolic_calcium_concentration')
+            return None
+
+    def _get_use_get_intracellular_calcium_concentration(self):
+        """ Get wyether or not the model has cytosolic_calcium_concentration"""
+        try:
+            self._model.get_symbol_by_ontology_term(self._OXMETA, "cytosolic_calcium_concentration")
+            return True
+        except KeyError:
+            return False
+
+    def _get_state_variables(self):
+        """ Sort the state variables, in similar order to pycml to prevent breaking existing code"""
+        return sorted(self._model.get_state_symbols(),
+                      key=lambda state_var: self._state_var_key_order(state_var))
+
+    def _get_membrane_stimulus_current(self):
+        """ Find the membrane_stimulus_current variable if it exists"""
+        try:
+            return self._model.get_symbol_by_ontology_term(self._OXMETA, self._STIMULUS_CURRENT)
+        except KeyError:
+            self._logger.debug(self._model.name + ' has no membrane_stimulus_current')
+            return None
+
+    def _get_membrane_stimulus_current_units(self):
+        """ Find the units for the membrane_stimulus_current variable if it exists"""
+        if self._membrane_stimulus_current is not None:
+            return self._model.units.summarise_units(self._membrane_stimulus_current)
+        else:
+            return None
+
+    def _get_membrane_capacitance(self):
+        """ Find membrane_capacitance if the model has it"""
+        if self._membrane_stimulus_current is not None:
+            # add membrane_capacitance if the model has it
+            try:
+                membrane_capacitance = self._model.get_symbol_by_ontology_term(self._OXMETA, "membrane_capacitance")
+                current_units = self._model.units.summarise_units(membrane_capacitance)
+                desired_units = getattr(self._model.units.ureg, self._STIMULUS_UNITS['membrane_capacitance'])
+                factor = self._model.units.get_conversion_factor(desired_units, from_unit=current_units)
+                initial_value = self._model.get_initial_value(membrane_capacitance)
+                equation = sp.Eq(membrane_capacitance, initial_value)
+                return equation, factor
+            except KeyError:
+                return None, 1.0
+
+    def _get_stimulus_currents(self):
```

Comment:
Not sure I get this. Is there every more than 1 stimulus current?

Reply:
hhm currents is maybe not the word. I'll just call it _get_stimulus
result looks like this:

`"{'membrane_stimulus_current_amplitude': Eq(_membrane$stim_amplitude, -20.0), 'membrane_stimulus_current_duration': Eq(_membrane$stim_duration, 0.0002), _membrane$stim_end: Eq(_membrane$stim_end, 100.0), 'membrane_stimulus_current_period': Eq(_membrane$stim_period, 0.5), 'membrane_stimulus_current_offset': Eq(_membrane$stim_start, 0.01)}"`